### ISSUE TITLE
No captions button for captioned HLS live stream

### ISSUE THREAD

**Issue by ranuser99** [videojs/video.js#4300]
## Description

Captions button does not appear when viewing a live stream with captions. 

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.

Using videojs (5.19.2), Hls.js (0.7.5), video-contrib-hls.js (2.0.1), view any HLS stream with captions.

## Results
Videojs will not display captions button.

However captions are being processed and can be displayed by calling:

videojs.textTracks[0].mode = 'showing'

### Expected

Videojs should display captions button.

Or, is there a config parameter or setup procedure that must be implemented?

#### videojs
what version of videojs does this occur with?  5.19.2
#### browsers
what browser are affected? All.
#### OSes 
what platforms (operating systems and devices) are affected? All
### plugins
Hls.js 0.7.5
videojs-contrib-hls.js 2.0.1

**Comment by skyh** [comment]
@ranuser99 you can try to use version <= 5.13.1 - CC button works correctly (for me)

**Comment by gkatsev** [comment]
Is the item that is showing up called "unknown"? If so, that's a [known issue](https://github.com/videojs/video.js/issues/2808) and is really an [Apple limitation](https://developer.apple.com/library/content/qa/qa1801/_index.html)

**Comment by gkatsev** [comment]
Do you have a live example? Are the captions in-manifest? If so, it would probably be up to hljs and the videojs-contrib-hls.js to provide the captions to the browser/videojs.

**Comment by skyh** [comment]
> Are captions in-manifest?

No, there are no captions in the manifest file and no captions in `<track>` elements. I suppose captions are in *.ts file.

Live examples:

link | version | HAP plugin | result
--- | --- | --- | ---
https://jsbin.com/kawaxezivi | `6.0.0` | `vjs-contrib-hls` | no video 
https://jsbin.com/wapacexuzo | `5.13.2` | `hola_hap` | native tracks
https://jsbin.com/liqigiloyi | `5.19.2` | `hola_hap` | no CC button

**Comment by ranuser99** [comment]
Will check out 5.13.2 thank you.

Does videojs utilize native caption drawing when available?  Can this be controlled at config level?

For some live streams, I have found drawing issues in the videojs-contrib-hls demo, whereas the hls.js demo which works with the native html5 video element, displays perfectly.  Example:
http://streambox.fr/mse/hls.js-0.7.6/demo/
using
http://playertest.longtailvideo.com/adaptive/captions/playlist.m3u8

In Chrome/Firefox for example, you'll notice better text render quality, positioning and on other live streams that roll-on, their implementation is ideal.

Can native caption rendering be leveraged when available?  How so?

Thank you!

**Comment by gkatsev** [comment]
It looks like our move to always using emulated text tracks in 5.14.0 may be the issue here.
We have an option of forcing native tracks: `{html5: {nativeTextTracks: true}}` but it seems to not be working right now.

If I use latest videojs and latest contrib-hls (with a different stream) it does work there: https://jsbin.com/qizuzayizu/1/edit?html,js,output

I'm not sure why your stream doesn't play in contrib-hls.

Also, we probably should fix it so this use-case still works, at least if the flag mentioned above is set.

**Comment by ranuser99** [comment]
Yes ability to force native handling would be great.  

Do you know which version (if any)  `{html5: {nativeTextTracks: true}} ` was implemented?

**Comment by taylornewton** [comment]
I encountered the same problem using hls.js with the videojs-hls library. We are delivering VTT captions via hls.

In the Html5 tech, there is a supportsNativeTextTracks function that is currently only returning true if IS_SAFARI is true. This is outdated, since most major browsers now support VTT. I overwrote this function in my local copy of video.js to always return true, and the button is now appearing in Chrome, FF, IE, and Safari.

This function should optimally detect native caption support instead of hard coding the result based on browser.

**Comment by jpatreal** [comment]
What was the solution for this?
